### PR TITLE
LP-2843 Integrate Arabic Certificate content

### PR DESCRIPTION
**Pull request by asadali145**
[LP-2843](https://philanthropyu.atlassian.net/browse/LP-2843)

[Theme PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/540)

***

#### Description:

- Add custom methods to customise the Arabic certificates.
- Moved a helper from mako to backend.

### CODE REVIEW

**Comment by asadali145** [comment]
Intentionally splitted tests into 2 for simplicity.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3028/workflows/workflows_id/jobs/3028/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/certificates_override/constants.py (100%)
openedx/adg/lms/certificates_override/helpers.py (100%)
openedx/adg/lms/certificates_override/tests/test_helpers.py (100%)
-------------
Total:   87 lines
Missing: 0 lines
Coverage: 100%
-------------
```

File: openedx/adg/lms/certificates_override/helpers.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from openedx.core.djangoapps.certificates.api import display_date_for_certificate
+
+
+def get_arabic_certificate_extra_context(request, course_id):
+    """
+    Compute `date_issued` and `gender` for the Arabic Course Certificates.
+
+    This method is used directly from certificate template. This is done to avoid core changes and customise the
+    Arabic certificates. This method customised formatted date issued of certificate and gender to the user
+    to display custom messages.
```
Comment (annas-usman) [inline openedx/adg/lms/certificates_override/helpers.py:22]: Please go through the third sentence, it needs some reformatting, it's difficult to comprehend

File: openedx/adg/lms/certificates_override/tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    with translation.override(ARABIC_LANGUAGE_CODE):
+        expected_date = _(expected_date_format).format(  # pylint: disable=translation-of-non-string
+            month=strftime_localized(today, '%B'),
+            day=today.day,
+            year=today.year
+        )
+
+    assert actual_date_issued == expected_date
+    assert actual_gender == user_with_profile.profile.gender
```
Comment (annas-usman) [inline openedx/adg/lms/certificates_override/tests/test_helpers.py:107]: We can combine the two test cases as most of the code is being repeated on the basis of user type.
```
if super_user condition:
    kwargs = {'course_id': course.id}
    preview_url = reverse('certificates:preview_cert', kwargs={'course_id': course.id})
    preview_url = f'{preview_url}?preview=honor'

    request = RequestFactory().get(preview_url)
    request.user = super_user
else:
    certificate = GeneratedCertificateFactory(
        user=user_with_profile,
        course_id=course.id,
        status=CertificateStatuses.downloadable
    )

    kwargs = {'certificate_uuid': certificate.verify_uuid}
    request = RequestFactory().get(reverse('certificates:render_cert_by_uuid', kwargs=kwargs))

**Review by annas-usman — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/certificates_override/tests/test_helpers.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    Test that 'set_certificate_recipient_font_size' returns expected font-size for a specific length string.
+    """
+    actual_font_size = set_certificate_recipient_font_size(accomplishment_copy_name)
+    assert actual_font_size == expected
+
+
+@pytest.mark.django_db
+def test_get_arabic_certificate_extra_context_existing_certificate(user_with_profile, mocker):
+    """
+    Test that `get_arabic_certificate_extra_context` return expected extra context for existing certificates.
```
Comment (annas-usman) [inline openedx/adg/lms/certificates_override/tests/test_helpers.py:39]: returns*

File: openedx/adg/lms/certificates_override/tests/test_helpers.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        )
+
+    assert actual_date_issued == expected_date
+    assert actual_gender == user_with_profile.profile.gender
+
+
+@pytest.mark.django_db
+def test_get_arabic_certificate_extra_context_preview_certificate(user_with_profile, mocker, super_user):
+    """
+    Test that `get_arabic_certificate_extra_context` return expected extra context for preview certificate.
```
Comment (annas-usman) [inline openedx/adg/lms/certificates_override/tests/test_helpers.py:77]: returns*

**Review by annas-usman — APPROVED** [review]
LGTM

File: openedx/adg/lms/certificates_override/helpers.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    Args:
+        accomplishment_copy_name (str): Name to be displayed on certificate.
+
+    Returns:
+        str: Font size.
+    """
+    size = len(accomplishment_copy_name)
+
+    if size > 32:
+        return '20px'
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/certificates_override/helpers.py:73]: Can we please define constants for pixel sizes?

File: openedx/adg/lms/certificates_override/tests/test_helpers.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    kwargs = {'certificate_uuid': certificate.verify_uuid}
+    request = RequestFactory().get(reverse('certificates:render_cert_by_uuid', kwargs=kwargs))
+
+    mock_resolver_match = mocker.patch.object(request, 'resolver_match')
+    mock_resolver_match.kwargs = kwargs
+
+    actual_gender, actual_date_issued = get_arabic_certificate_extra_context(request, course.id)
+
+    expected_date_format = u'{day} {month} {year}'
+    today = now()
```
Comment (javeriaejaz-arbisoft) [inline openedx/adg/lms/certificates_override/tests/test_helpers.py:61]: There is a fixture for the current date.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3033/workflows/workflows_id/jobs/3033/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/certificates_override/constants.py (100%)
openedx/adg/lms/certificates_override/helpers.py (100%)
openedx/adg/lms/certificates_override/tests/test_helpers.py (100%)
-------------
Total:   87 lines
Missing: 0 lines
Coverage: 100%
-------------
```

File: openedx/adg/lms/certificates_override/tests/test_helpers.py (hunk 2 above)
Comment (asadali145) [reply openedx/adg/lms/certificates_override/tests/test_helpers.py]: @javeriaejaz-arbisoft as I replied to Anas, The similarities are mocking and assertion. The core code is testing different views. If you still think that we  should merge these then I can. Actually initially I created 1 test and used parametrization but then splitted that test intentionally.

File: openedx/adg/lms/certificates_override/helpers.py (hunk 5 above)
Comment (asadali145) [reply openedx/adg/lms/certificates_override/helpers.py:73]: I don't think there is any need to create constants for this one, Also this is moved from mako to backend.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3034/workflows/workflows_id/jobs/3034/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/certificates_override/constants.py (100%)
openedx/adg/lms/certificates_override/helpers.py (100%)
openedx/adg/lms/certificates_override/tests/test_helpers.py (100%)
-------------
Total:   82 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Pull request merged**